### DISCUSSION TITLE
plugin for server

### DISCUSSION THREAD

**Discussion by Derxibitor** [Pryaxis/TShock#2473] [Q&A]
I want to play with my friends, one of my friend play on mobile phone. I started a server with plugin "crossplay" it seems that it doesnt work as i want. i am convienced that i download the plugin coorrctly, because the Tshock manager wrote "crossplay plugin initiated". My friend that played on computer could connect to my server and my friend that played on mobile phone didnt. Terraria version is 1.4.0.5

plugin i installed https://github.com/Moneylover3246/Crossplay/releases/tag/8bac2da

can somebody help me?

**Comment by Moneylover3246** [comment]
Hello, I've noticed you've already created an issue about this a couple of hours ago. Other than that here's a couple of steps to make sure your friend can join properly.

- The IP you're giving him isn't `192.168.x.x` or `127.0.0.1`
- You've port forwarded, and to the correct port
- Your firewall allows connections through said port

Are all these steps correctly done? If you need help with them make sure to let it be known

**Reply by Derxibitor** [reply]
how i can check my firewall?

**Comment by Moneylover3246** [comment]
Did you check for the correct ip and that you port forwarded? If so, what operating system sre yoy using?

**Reply by Derxibitor** [reply]
dude, i wrote, my another friend could connect. My only friend that played mobile didnt

**Reply by Derxibitor** [reply]
> Did you check for the correct ip and that you port forwarded? If so, what operating system sre yoy using?

i also noticed that you post crossplay terraria plugin, can you please say, does it works with mobile phones? i mean if i host a server on pc, mobile phone user can connect to the server?

**Reply by Moneylover3246** [reply]
The problem isn't with the functionality of the plugin. If your friend is stuck on the connecting screen that means you're giving them the wrong input somewhere

**Reply by Derxibitor** [reply]
> The problem isn't with the functionality of the plugin. If your friend is stuck on the connecting screen that means you're giving them the wrong input somewhere

I am sure that i gave my friend correct IP address, because my another friend connected by this IP address. Just to remind you, my friend that plays on PC can connect, but my friend that plays on mobile phone cannot connect. I think plugin doesnt work with mobile phones or some how i dont install this correct. I know that you post this plugin, can you please admit that it works with mobile phones? If yes, what i did wrong? Tshock wrote that the plugin initiated.

**Reply by Moneylover3246** [reply]
Regardless of the current version of the server, all mobile players will at least try to make an attempt to connect to it. Normally, they'll be kicked once their version is figured out to not match PC's though

All the crossplay plugin does is bypass this restriction and allow them to play in the server normally. This plugin works (mostly) correctly. You should try to bring this up with your friend so you can fix whatever mistakes they're having

**Reply by Moneylover3246** [reply]
If you truly need help with this, I may be able to assist you better by joining Pryaxis' discord [here](https://discord.gg/Cav9nYX)
